crates/router/src/core/payments/operations/payment_create.rs

```diff
@@ -1217,13 +1217,22 @@ async fn create_payment_link(
 ) -> RouterResult<Option<api_models::payments::PaymentLinkResponse>> {
     let created_at @ last_modified_at = Some(common_utils::date_time::now());
     let payment_link_id = utils::generate_id(consts::ID_LENGTH, "plink");
-    let payment_link = format!(
+    let open_payment_link = format!(
         "{}/payment_link/{}/{}",
         domain_name,
         merchant_id.clone(),
         payment_id.clone()
     );
 
+    let link_secure = payment_link_config.allowed_domains.as_ref().map(|_| {
+        format!(
+            "{}/payment_link/s/{}/{}",
+            domain_name,
+            merchant_id.clone(),
+            payment_id.clone()
+        )
+    });
```

Comment:
what is `s` in the path param, why not explicitly state `secure`

Reply:
https://github.com/juspay/hyperswitch/pull/5357#discussion_r1690799908